### DISCUSSION TITLE
Macbook air vale a pena para front-end ?

### DISCUSSION THREAD

**Discussion by Juuhz** [frontendbr/forum#1329] [Perguntas e Respostas]
Pessoal, tudo bem ?

Estou em busca do meu primeiro notebook, sempre tive desktops e estou com algumas dúvidas na compra, gostaria de saber das experiências de vocês. 

Estou de olho em um **Macbook Air** na Americanas, está **4500** mas tem quase **700 reais** de cashback ( que está me atentando bastante ), mas achei as configurações de hardware baixas.

https://www.americanas.com.br/produto/132490742/macbook-air-mqd32bz-a-com-intel-core-i5-dual-core-8gb-128gb-ssd-13-prata-apple?WT.srch=1&epar=bp_pl_00_go_inf_notebooks_todas_geral_gmv&gclid=Cj0KCQiAsJfhBRCaARIsAO68ZM4d1KdhRY-RWN0Ca2j5lUFm8Ph2VnfgYBd1DmOOhu9vAS0cJZZtHRgaAkseEALw_wcB&opn=YSMESP&sellerId=00776574000660

## Dúvidas:

1) Rodaria de boa o setup de front-end ? ( IDE / Navegador / Terminal npm / etc... )
2) Mesmo esse modelo sendo de 2013, vale a pena ?
3) Ou é melhor pegar um **pro** usado no mercado livre ? ( pelo mesmo valor ).
4) Ou é melhor comprar um notebook windows mais potente ( em termos de hardware ), que sairia mais barato e eu poderia instalar o linux ? Se sim, qual ?

5) Sugestões ? 
6) Qual notebook vocês utilizam ? Pagaram quanto ?

Abraços.

Feliz Ano Novo <3

**Comment by luizbills** [comment]
Com um Dell por esse valor você ganha o dobro de hardware ou até mais.

**Comment by vagnerlandio** [comment]
Verdade, pelo mesmo preço você consegue um Dell ou Asus com Intel Core i7 e 16GB de RAM e com hardwares atuais.

**Comment by jhonattan-santos** [comment]
Comprei um pro 2012 i7 120SSD e 8GB e tem atendido bem as minhas espectativas, usei muito windows, depois 3 anos de linux e à alguns meses o osx e na minha opnião está valendo a pena, tenho um thinkpad i5 6a geração com 12GB de ram e o mac tem tido um desempenho legal comparado ao thinkpad. 

Pra mim que desenvolvo e faço trabalhos de designer valeu a pena a quisição. Rodando Mojave aqui e o Time Machine me poupa bastante tempo quando eu faço cagada no sistema, toda vez que eu vou fazer algo que pode ser perigoso ou posso avacalhar minhas configurações eu salvo no HD externo com o time machine e se der problema, restauro como estava anteriormente com todos os apps instalados e configurações anteriores em pouco tempo.

**Comment by Sergioamjr** [comment]
No meio do ano eu comprei um macbook air 2017 e tem super valido a pena. Uso pra estudo, trabalho. Era o mais barato entre os macs mas ainda assim é uma ótima máquina. Raramente trava, tem compatibilidade com tudo, rápido, leve, além da durabilidade de anos, coisa que eu não posso garantir com um Dell e windows. Paguei 4,300 no Mercado Livre com NFe, garantia e tudo.

**Comment by marcusvcribeiro** [comment]
Vão falar um monte de baboseira ai dizendo que consegue configurações
melhores com outros computadores.
Porem esteja ciente que todo macbook tem configurações inferiores a outros
notebooks, só que o Sistema Operacional faz total diferença.
Um macbook 2013, I5, 256 SSD bate facil um I7,Placa de video +SSD.

Depois leia sobre Softwares feitos para rodar em sistemas genericos e
softwares feitos para rodar em um unico hardware, e veja a diferença.

Esse Macbook ai te atende perfeitamente para qualquer coisa de frontend.
Porem eu pegaria um 2015 usado, só procurar um pouco que voce acha.


Em seg, 31 de dez de 2018 às 18:02, Sérgio Junior <notifications@github.com>
escreveu:

**Comment by caioincau** [comment]
Atende de boa sim, porém, os 8GB de ram vão limitar a vida dele para um ou dois anos, assim como o que eu comprei(Pro 2015, i5, 8GB, 240SSD).

8GB de RAM com Chrome + Node + VSCode + Spotify funciona, mas fica no limite.

**Comment by willycamargo** [comment]
Pode ajudar:
#817  
#1285

**Comment by jampow** [comment]
Trabalho como front-end e preciso subir uma stack com 15 containers docker para simular o ambiente na máquina local. o Mac com 8GB não deu conta, já no linux foi bem suave. Se vc precisar de alguns containers pode ter dificuldades.

**Comment by andresantos-tech** [comment]
Não sei se a experiência vale porque trabalho mais com backend, mas trabalhei com um macbook pro por uns 5 ou 6 anos, e mudei pelo motivo que o @jampow citou. Mesmo com upgrades o gargalo acabou sendo o processador.

Se for comprar um macbook mesmo, só fique atendo que os modelos pós 2012 não permitem upgrade de memória ram e hd/ssd (pelo menos não de um jeito fácil), então é bom escolher uma configuração que já te atenda por um bom tempo.

**Comment by MRRaduan** [comment]
Sou Front e atualmente no meu trabalho estou usando um MacBook Air 2017, i5 com 8GB de RAM. 
Já me aventurei trabalhando usando Windows e pessoalmente uso Linux.

**O fato:**
O mac ta dando conta do recado. As vezes preciso abrir alguns arquivos pesados do Photoshop e ultimamente ando abrindo mais alguns arquivos no Adobe Xd e tudo numa boa. 
De processamento para desenvolver é mais tranquilo ainda, pelo menos para a atual stack que tenho utilizado. 

**Agora minhas considerações pessoais:**
Fora do trabalho eu sou usuário linux, atualmente estou utilizando o Deepin.
Meu computador pessoal é um notebook Asus X555UB, i5, 1TB, 2GB GeForce e recentemente fiz um upgrade de 256GB de SSD, e cara.. Em relação a desempenho, é tudo a mesma coisa. Meu notebook Asus tem o mesmo desempenho que o meu Mac Air custando menos da metade do preço adicionando o preço da SSD. Pelo menos até o momento está suprindo minhas necessidades e demandas numa boa.

**As reais diferenças**
_Tela:_ a imagem do mac é realmente muito boa, mesmo não sendo retina eu fiquei satisfeito.
A resolução da tela é mediana, ainda mais por ser uma tela pequena fica bem limitado o aproveitamento de tela para trabalho.

_A carcaça_: O Mac é todo feito em alumínio e passa a sensação de ser um aparelho resistente e que durará.

_O design_: Realmente é muito bonito.

_A bateria_: Esse sim eu posso dizer que é a maior vantagem de todas: a bateria do mac, comparado aos notebook convencionais que vemos por aí, dura muito mesmo.

_Portabilidade_: Ele é pequeno, fino e relativamente leve, o que facilita muito qualquer locomoção, seja pra uma reunião ou para viagem.

Mas nada que um ThinkPad T470 (que tem um desempenho de duração da bateria enorme e teclado mecânico) ou algum Dell novo da serie Inspiron (que tem uma tela IPS com a resolução Full HD muito melhor que a do mac, e parte da carcaça feita em alumínio ) ou qualquer outro notebook com uma configuração boa e uma SSD não deixa de fazer.

Boa parte do desempenho necessário das máquinas que utilizei para trabalho vem de um bom processador e uma SSD.

Espero ter ajudado, abraços.

**Comment by Juuhz** [comment]
Galera,

primeiramente, gostaria de agradecer a cada um de vocês que contribuíram de alguma forma para esse passo que eu acabei de dar ( sim, comprei o note ). Essa foi minha primeira issue/participação e me senti abraçado pela comunidade, vocês são demais! E com certeza eu irei contribuir também com minhas expertises. 

Bom, depois do feedback de cada um e acoplando as minhas necessidades, optei por continuar a dar preferência a linha Macbook, por N motivos:

- Sistema operacional.
- Hardware. 
- Bateria.
- Durabilidade.
- Publicação da Apple store.
- etc...

Eu dei uma garimpada e achei um Macbook Pro de 2015 com:

- Tela de Retina
- i5 de 2,7 Ghz
- 256 SSD
- 8GB de ram

Por 4,500 no mercado livre ( Usado ). 

Mas eu peguei um notebook recondicionado de uma apreensão ( aquelas que a polícia faz em empresas ), ou seja, peguei um notebook zerado com um descontão. 

Acredito que essas configurações com certeza irão suprir minhas necessidades como DEV, no trabalho já utilizo um Mac Mini inferior e já atende bem. 

Novamente, obrigado a todos <3

Segue a foto do meu novo BB:
![captura de tela 2019-01-03 as 16 27 24](https://user-images.githubusercontent.com/7996030/50654467-12f24100-0f74-11e9-9ba2-9b7b70b43521.png)

**Comment by rubenmarcus** [comment]
@Mendrone 

comprei um Dell Inspiron i14-7460-A20G  por 4k ano passado dei 1 upgrade nele com 1 hd M2 da samsung de 500gb (ficou 1tb hdd + 500gb m2) + 32gb de memória, e me arrependi.

Porque? gastei 6k num equipamento que eu esperava durar pelo menos 5 anos, e hoje 1 ano e meio depois, a bateria já apresenta 90% de desgaste.

Tudo bem que eu sou heavy user e deixo o note ligado como um desktop. Uso 2 monitores ligados nele, um pela porta hdmi e um por um adaptador usb que comprei.

Mas me arrependi. Tem uma porta usb que não reconhece os usbs, e a bateria ja foi pro saco em 1 ano e meio.

Além de dev tambem trabalho com fotografia como hobby, e photoshop fazendo artes digitais, ele aguentou bem em viagens que eu fiz, mas a perfomance da bateria de 2 meses pra cá caiu de 48% de desgaste pra 90%

ele ja perdeu o 1 ano de garantia, e a bala que paguei 1 ano e meio atras,  vou precisar pagar mais uns 800 reais pra comprar uma bateria nova.


A ventoinha dele faz muito barulho as vezes, e esquenta demais também.

Fora que o Windows trava as vezes o chrome, da umas engasgadas, pra jogos e apps ele roda liso, mas no chrome é facil o windows ficar com 100% de cpu.

Eu quero usar Linux na maquina, porque nao consigo rodar os shellscripts do meu trabalho. e o terminal do windows é uma merda.

**Comment by vitor-mendrone-deel** [comment]
@rubenmarcus Não sei se me marcou por acidente nessa issue ou se foi porque eu comprei o mesmo notebook a pouco mais de um ano e indiquei a máquina hahahaha Se foi pelo último motivo, peço perdão pelo vacilo! 

De qualquer forma minha percepção do mesmo é semelhante, incrível como com o mesmo processador e configurações semelhantes a do Mac em questões de Hardware ainda oferece muito menos performance e durabilidade. 

Tive a vantagem de ter pago bem pouco nele porque a empresa que eu trabalhava costumava comprar notebooks da Dell para o time de negócios e com isso, usando o desconto que eles tinham, eu paguei 2k na máquina. 

Porém hoje, após 1 ano e meio de uso, é perceptível a queda na performance. Não chega a ser um impeditivo pra o uso, porém tudo leva mais tempo.. inicializar, abrir um programa, abrir o terminal, rodar um `npm install` ou um `rm -rf` em uma pasta com muita coisa é um pesadelo. 

Hoje eu diria a qualquer um que queira um notebook pra trabalho, se tiver condições, fazer o esforço de comprar um Mac, novo ou usado, é **ridiculamente** melhor.

**Comment by yuritoledo** [comment]
@Juuhz tem o link para essas aquisições de caloteiros? :D

**Comment by Juuhz** [comment]
Tenho sim:

https://produto.mercadolivre.com.br/MLB-1120221545-macbook-pro-2014-retina-i5-8gb-256ssd-carregador-garantia-_JM?quantity=1

Eu peguei a unidade 2, que não tinha amassado e estava com 90 ciclos na bateria ( novissimo ). Agora já tem outra anunciada no lugar.

Recomendo esse vendedor, super atencioso, chegou em menos de 1 semana o produto.

**Comment by yuritoledo** [comment]
@Juuhz caramba, que bacana man

Deu sorte mesmo.

Eu preciso de um, por mexer com react-native, pra gerar os apps :(

**Comment by Juuhz** [comment]
@yuritoledo,

Super recomendo cara, da uma procurada bacana que você acha, precisar de ajuda, tamo aê.

**Comment by yuritoledo** [comment]
@Juuhz Valeu!

**Comment by expalmer** [comment]
Aproveitando a thread, eu uso um mac book air 2017 
1.8 GHz Intel Core i5
8 GB 1600 MHz DDR3
Gente, só que ele tem 128G, e só sistema operacional está ocupando 100GB e 10GB de APP, não tenho mais espaço!
Alguém sabe como remover algum arquivo de sistema desnecessário, logs, arquivos antigos de instalação ?
Ou será que tem como colocar mais de HD ?
